.github/workflows/docs.yml

```diff
@@ -93,6 +93,16 @@ jobs:
         if: always()
         run: if [ -e doc/errors.txt ]; then cat doc/errors.txt; fi
 
+      - name: Test Documentation Images
+        run: pytest tests/doc/tst_doc_images.py
+
+      - name: Upload Test Debug Images
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: doc-debug-images
+          path: _doc_debug_images
+
```

Comment:
@user27182 Should we also download the artifact and run the test? Or would you prefer to upload only the artifact and allow the user to test locally? I believe that it will lead to different ways of solving CI errors.

Reply:
Didn't see this earlier, can you clarify what you mean? Are you wondering if a user can reproduce the same tests locally that were done with CI?

Reply:
Sorry, I can't remember what I was going to say. Please give me time to recognize it 😅

Reply:
@user27182 Regarding this thread, I meant I need advice on the procedure for merging this PR. How should I do it?

Reply:
Ah ok. I would download the doc_debug_images artifact from the latest failed build. Copy them to the your local pyvista repo in the doc_image_cache directory, and push a commit with the new images. 

This should be the expected procedure to follow when adding a new docstring example plot to be included in the docs, for example. 

At first the build will fail, and a helpful error should be generated by the tests. It should be clear that a cached image is missing and should be added. 

This should be documented in CONTRIBUTING. 

If you'd like I can commit the cache images.

Reply:
Maybe it would be helpful if the artifact only included the missing images? And not the entire cache

Reply:
> Maybe it would be helpful if the artifact only included the missing images? And not the entire cache

This is only helpful for some cases. But the tests can fail for missing cache images, but also for missing doc images or for regression errors. So, a separate artifact with missing cache images only may not always be helpful. I am not sure it's worth the added complexity to add an artifact or copy the images for failed tests into a separate folder.

The failure message I think are clear enough to easily find the image you need from all the debug images in the artifact. And it's less than 80MB to download.

Reply:
I did find it was annoying to try to find a single file from inside the debug image dir with 2000+ images. With https://github.com/pyvista/pyvista/pull/5718/commits/c2f1093c69f8ab277843cd26e0f9b5bc191ef37e and https://github.com/pyvista/pyvista/pull/5718/commits/49ff47922bfea59f66c8d47c22cdc03653f48b5d a separate failed-cases only artifact is uploaded